backend/donor-service/main.js

```diff
@@ -570,7 +584,21 @@ app.post('/auth/mobile/verifyOTP', async(req, res) => {
         console.debug('OTP verified', verifyOtpResponse);
         if (R.pathOr([], ["mobileLinkedHid"], verifyOtpResponse).length > 0) {
             for (const data of verifyOtpResponse.mobileLinkedHid) {
-                data["pledged"] = await isABHARegistered(data.healthIdNumber)
+                let abhaRegistered = await isABHARegistered(data.healthIdNumber, true)
+                data["pledged"] = abhaRegistered
+                if (abhaRegistered) {
+                    let pledgeStatus = await getPledgeStatus(data.healthIdNumber)
+                    switch(pledgeStatus) {
+                        case PLEDGE_STATUS.PLEDGED : 
+                            data["pledgeStatus"] = pledgeStatus
+                            break;
+                        case PLEDGE_STATUS.UNPLEDGED: 
+                            data["pledgeStatus"] = pledgeStatus
+                            break;     
+                    } 
```

Comment:
why do we need this switch?

Reply:
This I added at first to handle the exiting case of value being stored as **_true_** for a given D+ ABHA key. But then made changes and might have missed the logic that in both cases it's the same value being retuned. But now I have made changes accordingly considering all scenarios. And the switch case is being used appropriately